scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -119,15 +119,17 @@ object Term {
   }
   @ast class Try(expr: Term, catchp: List[Case], finallyp: Option[Term]) extends Term
   @ast class TryWithHandler(expr: Term, catchp: Term, finallyp: Option[Term]) extends Term
-  @ast class ContextFunction(params: List[Term.Param], body: Term) extends Term {
+
+  @branch trait FunctionLike extends Term
+  @ast class ContextFunction(params: List[Term.Param], body: Term) extends FunctionLike {
```

Comment:
Unfortunately, we would need to fix the macros and I would rather avoid it since usually I spend at least a day dealing with it.

Reply:
can i try to help? is `@branch` required?

Reply:
`@branch` is required otherwise it complains about it not being part of the ast. If I try to add the methods into the branch annotated trait it complains that it needs to be abstract. I think this is because the macros generate an additional class for `@branch` traits which is why that error is shown.

I also tried to implement them with `???`, but the it seems some of the `@ast` generated classes don't get the override modifier.

 I could try to work on it, but I am afraid it's a rabbit whole I will not come out of soon :sweat_smile:

Reply:
Or alternatively to not play with macros we could create an implicit class for it and add `params` and `body` as extension methods. What do you think? @kitbellew

Reply:
Now that #2211 has been merged, you can change this to:
```
  @branch trait FunctionLike extends Term {
    def params: List[Term.Param]
    def body: Term
  }
```
and change two more files:
```
// tests/api/SurfaceSuite.scala:334
      |scala.meta.Term.FunctionLike

// tests/trees/ReflectionSuite:23
    assertEquals(symbolOf[scala.meta.Tree].asRoot.allBranches.length, 20)
    assertEquals(symbolOf[scala.meta.Tree].asRoot.allLeafs.length, 338)
```